scrapy/downloadermiddlewares/retry.py

```diff
@@ -23,12 +24,104 @@
 )
 from twisted.web.client import ResponseFailed
 
-from scrapy.exceptions import NotConfigured
-from scrapy.utils.response import response_status_message
 from scrapy.core.downloader.handlers.http11 import TunnelError
+from scrapy.exceptions import NotConfigured
+from scrapy.http.request import Request
+from scrapy.spiders import Spider
 from scrapy.utils.python import global_object_name
+from scrapy.utils.response import response_status_message
+
 
-logger = logging.getLogger(__name__)
+retry_logger = getLogger(__name__)
+
+
+def get_retry_request(
+    request: Request,
+    *,
+    spider: Spider,
+    reason: Union[str, Exception] = 'unspecified',
+    max_retry_times: Optional[int] = None,
+    priority_adjust: Union[int, float, None] = None,
```

Comment:
can it be float?

Reply:
`Request.priority` [can only be int](https://github.com/scrapy/scrapy/blob/master/scrapy/http/request/__init__.py#L27-L28), `new_request.priority = request.priority + priority_adjust` would make it a float if `priority_adjust` were a float. Updated at d458ccff3b2d6df94df1aa86eeb7d2505d62f2d6.

/cc @wRAR